saleor/warehouse/migrations/0003_warehouse_slug.py

```diff
@@ -1,37 +1,26 @@
 # Generated by Django 2.2.9 on 2020-01-29 06:52
 
-from django.db import migrations, models
+from django.db import migrations, models, transaction
+from django.db.utils import IntegrityError
 from django.utils.text import slugify
 
 
 def create_unique_slug_for_warehouses(apps, schema_editor):
     Warehouse = apps.get_model("warehouse", "Warehouse")
 
-    for warehouse in Warehouse.objects.all():
-        if not warehouse.slug:
-            warehouse.slug = generate_unique_slug(warehouse)
-            warehouse.save(update_fields=["slug"])
-
-
-def generate_unique_slug(instance):
-    slug = slugify(instance.name)
-    unique_slug = slug
-
-    ModelClass = instance.__class__
-    extension = 1
-
-    pattern = rf"{slug}-\d+$|{slug}$"
-    slug_values = (
-        ModelClass._default_manager.filter(slug__iregex=pattern)
-        .exclude(pk=instance.pk)
-        .values_list("slug", flat=True)
-    )
-
-    while unique_slug in slug_values:
-        extension += 1
-        unique_slug = f"{slug}-{extension}"
-
-    return unique_slug
+    for warehouse in Warehouse.objects.filter(slug__isnull=True):
+        slug = slugify(warehouse.name)
+        extension = 1
+        slug_value = slug
+        while True:
+            try:
+                with transaction.atomic():
+                    warehouse.slug = slug_value
+                    warehouse.save(update_fields=["slug"])
```

Comment:
Not needed. Wrap `create_unique_slug_for_warehouses` into an atomic transaction instead

Reply:
I changed soultion, there is no more transaction.